src/h5web/dataset-visualizer/DimensionMapper.tsx

```diff
@@ -1,86 +1,78 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import ReactSlider from 'react-slider';
+import { isNumber } from 'lodash-es';
 import { Vis, DimensionMapping } from './models';
-import { useVisProps } from './VisProvider';
 import ButtonGroup from '../visualizations/shared/ButtonGroup';
 import styles from './DimensionMapper.module.css';
 
 interface Props {
   activeVis: Vis;
+  rawDims: number[];
+  mapperState: DimensionMapping;
   onChange: (d: DimensionMapping) => void;
 }
 
 function DimensionMapper(props: Props): JSX.Element {
-  const { activeVis, onChange } = props;
-  const { rawDims, mapping } = useVisProps();
+  const { activeVis, rawDims, mapperState, onChange } = props;
```

Comment:
As discussed, everything is fetched by props rather than with the hook. This is enforced as `DimensionMapper` has no longer access to the `VisContext`.